Sources/Common/DataModel/CardinalSpline1D/index.js

```diff
@@ -51,16 +53,14 @@ function vtkCardinalSpline1D(publicAPI, model) {
     model.coefficients[0 * 4 + 3] = 1;
 
     for (let k = 1; k <= N; k++) {
-      model.coefficients[4 * k + 1] -=
-        model.coefficients[4 * k + 0] * model.coefficients[4 * (k - 1) + 2];
-      model.coefficients[4 * k + 2] =
-        model.coefficients[4 * k + 2] / model.coefficients[4 * k + 1];
+      model.coefficients[k * 4 + 1] =
+        model.coefficients[k * 4 + 1] -
+        model.coefficients[k * 4 + 0] * model.coefficients[(k - 1) * 4 + 2];
+      model.coefficients[k * 4 + 2] =
+        model.coefficients[k * 4 + 2] / model.coefficients[k * 4 + 1];
       work[k] =
-        (work[k] - model.coefficients[4 * k + 0] * work[k - 1]) /
-        model.coefficients[4 * k + 1];
-      model.coefficients[4 * k + 3] =
```

Comment:
what were with those lines  ? did you fix a bug ?

Reply:
It was to make that part of the code for open and close closer to each other as both seemed to work. However I moved it back again to what it was to avoid any further problem and to make it closer to the C++ version.